src/components/periodic-report/periodic-report.service.ts

```diff
@@ -0,0 +1,306 @@
+          ...property('end', input.end, 'newPeriodicReport'),
+        ])
+        .return('newPeriodicReport.id as id');
+      const result = await createPeriodicReport.first();
+
+      if (!result) {
+        throw new ServerException('Failed to create a periodic report');
+      }
+
+      return await this.readOne(id, session);
+    } catch (exception) {
+      throw new ServerException('Could not create periodic report', exception);
+    }
+  }
+
+  async uploadFile(
+    reportId: string,
+    file: CreateDefinedFileVersionInput,
+    session: Session
+  ) {
+    const reportFileId = await generateId();
+
+    await this.files.createDefinedFile(
+      reportFileId,
+      file.name ?? 'Report File',
+      session,
+      reportId,
+      'reportFile',
+      file,
+      'periodicReport.reportFile'
+    );
+
+    await this.db
+      .query()
+      .match(node('periodicReport', 'PeriodicReport', { id: reportId }))
+      .match(node('file', 'File', { id: reportFileId }))
+      .create([
+        node('periodicReport'),
+        relation('out', '', 'reportFile', {
+          active: true,
+          createdAt: DateTime.local(),
+        }),
+        node('file'),
+      ])
+      .run();
+
+    return await this.files.resolveDefinedFile(
+      {
+        value: reportFileId,
+        canRead: true,
+        canEdit: true,
+      },
+      session
+    );
+  }
+
+  async readOne(id: string, session: Session): Promise<PeriodicReport> {
+    this.logger.debug(`read one`, {
+      id,
+      userId: session.userId,
+    });
+    if (!id) {
+      throw new NotFoundException(
+        'No periodic report id to search for',
+        'periodicReport.id'
+      );
+    }
+
+    const query = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'PeriodicReport', { id })])
+      .call(matchPropList)
+      .optionalMatch([
+        node('node'),
+        relation('out', '', 'reportFile', { active: true }),
+        node('reportFile', 'File'),
+      ])
+      .return('node, propList, reportFile.id as reportFileId')
+      .asResult<
+        StandardReadResult<DbPropsOfDto<PeriodicReport>> & {
+          reportFileId: string;
+        }
+      >();
+
+    const result = await query.first();
+    if (!result) {
+      throw new NotFoundException(
+        'Could not find periodic report',
+        'periodicReport.id'
+      );
+    }
+
+    const props = parsePropList(result.propList);
+    const securedProps = await this.authorizationService.secureProperties(
+      IPeriodicReport,
+      props,
+      session
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...securedProps,
+      type: props.type,
+      start: props.start,
+      end: props.end,
+      reportFile: await this.files.resolveDefinedFile(
+        {
+          value: result.reportFileId,
+          canEdit: true,
+          canRead: true,
+        },
+        session
+      ),
```

Comment:
If we cannot grab the relation object in the same db query then we should defer that to another resolver. It allows this db query to be skipped entirely if `reportFile` is not asked for. 
See these examples:
https://github.com/SeedCompany/cord-api-v3/blob/bcfef17a3c44066d8f5f247825f9e5035f18621e/src/components/engagement/dto/engagement.dto.ts#L45-L45
https://github.com/SeedCompany/cord-api-v3/blob/bcfef17a3c44066d8f5f247825f9e5035f18621e/src/components/engagement/engagement.resolver.ts#L65-L76